rfc14: update definition of duration=0 (#315)
flux-framework/rfc: spec_14.rst

Some common system attributes are:

**duration**
   The value of the ``duration`` attribute is a floating-point number greater
   than or equal to zero representing time span in seconds.  The scheduler

>    will make an effort to allocate the requested resources for the number

Reviewer: I know this is not your change. But is there any case where the scheduler will not allocate for duration? If not, perhaps this can be a stronger requirement? 

```suggestion
SHALL allocate the requested resources for the number
```
This seems more aligned with the spirit of this change which makes the scheduler's behavior more explicit -- as a  requirement. 

I understand this may try to cover a case where the scheduler may not be able to allocate the duration at all. IMHO, we can assume that the scheduler can allocate in this section?

Author: Yes, that seems reasonable. I will make that change.

Author: Ok, I pushed the requested change, and also added the qualifier `If the duration is greater than zero` before that statement.

Reviewer: Yup better!
